### PR TITLE
upgrade v2

### PR DESCRIPTION
**Pull request by asalzmann**
## What is the purpose of the change

Currently, `stakeibc`'s module account is blacklisted, which has halted transfers. See #206 for more context and a long-term solution.,

This upgrade removes `stakeibc`s module account from the bank keeper blacklist. This is meant to be a temporary patch that we'll revert in the future.

## Brief Changelog

- add v2 upgrade scaffolding
- remove `stakeibc` module account from blacklist

## Testing and Verifying

Force a transfer timeout with the old blacklist
```
2022-08-22T21:56:30.742840Z ERROR ThreadId(23) send_tx_with_account_sequence_retry{id=STRIDE}:estimate_gas: failed to simulate tx. propagating error to caller: gRPC call failed with status: status: Unknown, message: "recovered: unable to send coins from module to account despite previously minting coins to module account: stride1mvdq4nlupl39243qjz7sds5ez3rl9mnx253lza is not allowed to receive funds: unauthorized [cosmos/cosmos-sdk@v0.45.5/x/bank/keeper/keeper.go:319]\nstack:\ngoroutine 43700 [running]:\nruntime/debug.Stack()\n\truntime/debug/stack.go:24 +0x65\ngithub.com/cosmos/cosmos-sdk/baseapp.n...
```

Force a transfer timeout with the new app.go blacklist. No errors! (except for the standard timeout)

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? na
  - Does this pull request update existing proto field values (and require a backend and frontend migration)? no
  - Does this pull request change existing proto field names (and require a frontend migration)? no

### CODE REVIEW

File: app/upgrades/v2/upgrades.go [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+	UpgradeName = "Upgrade to Resolve Consensus Bug"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v2
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		vm[recordstypes.ModuleName] = 2
```
Comment (sampocs) [inline app/upgrades/v2/upgrades.go:21]: I don't think we want to increment the records module consensus version here (just remove this line). I believe that's only if there's a state migration. 

And for future reference, when specifying a new version here, you also have to increment the `ConsensusVersion` in the module's `module.go` and add [migration handlers](https://github.com/Stride-Labs/stride/tree/main/app/upgrades#add-migration-handler)

**Review by sampocs — COMMENTED** [review]
Blacklisted validators code looks good! Couple notes on the upgrade logic

**Comment by sampocs** [comment]
We also need to register the upgrade handler in `app/upgrades.go`
```go
func (app *StrideApp) setupUpgradeHandlers() {
	// v2 upgrade handler
	app.UpgradeKeeper.SetUpgradeHandler(
		v2.UpgradeName,
		v2.CreateUpgradeHandler(app.mm, app.configurator),
	)

```
And update the version in `cmd/strided/main.go`

File: app/upgrades/v2/upgrades.go [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+
+	recordstypes "github.com/Stride-Labs/stride/x/records/types"
+)
+
+const (
+	UpgradeName = "Upgrade to Resolve Consensus Bug"
```
[the remark this answers is not in the document]
Comment (asalzmann) [reply app/upgrades/v2/upgrades.go:12]: done

File: scripts-local/test_upgrade.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (asalzmann) [reply scripts-local/test_upgrade.sh:1]: I'm not sure, I think @vish-stride would know!

**Review by vish-stride — APPROVED** [review]
looks great!

**Pull request merged**